Lesson2.ipynb

```diff
@@ -422,13 +460,134 @@
     "print(\"The last character of helloWorld is\", helloWorld[len(helloWorld)-1]) "
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Casting\n",
+    "\n",
+    "Are `1`, `1.0`, and `\"1\"` the same?\n",
+    "\n",
+    "`1` is an Integer, `1.0` is a Float and `\"1\"` is a String. But, from a human perspective, the three of them just represent the number 1.\n",
+    "\n",
+    "Let's find out if Python considers them equal:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(1==\"1\")\n",
+    "print(1==1.0)\n",
+    "print(1.0==\"1\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see by running the code cell above, Python allows you to compare numbers (1==1.0 is True), but you can't compare numbers and Strings together (1.0==\"1\" is False) !\n",
+    "\n",
+    "This means, when you want to compare data, you must make sure all the data is from the same type. To do so, you need to convert data from a Type to another. This is called **Casting**.\n",
+    "\n",
+    "In order to cast a value or a variable to a specific Type, you wrap them around the following methods:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "intFromFloat = int(1.0)\n",
+    "intFromString = int(\"1\")\n",
+    "print(\"1.0 and \\\"1\\\" are now both Integers:\", intFromFloat==intFromString) \n",
+    "\n",
+    "floatFromInt = float(1)\n",
+    "floatFromString = float(\"1\")\n",
+    "print(\"1 and \\\"1\\\" are now both Floats:\", floatFromInt==floatFromString) \n",
+    "\n",
+    "stringFromInt = str(1)\n",
+    "stringFromFloat = str(1.0)\n",
+    "print(\"1 and 1.0 are now both Strings. But are they equal?\", stringFromInt==stringFromFloat) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Hey?! Why the two Strings above are not equal??\n",
```

Comment:
"Why the two Strings above are not equal??" should read "Why are the two Strings above not equal??"